packages/firestore/src/api/timestamp.ts

```diff
@@ -92,4 +89,31 @@ export class Timestamp {
       ')'
     );
   }
+
+  // Overriding valueOf() allows Timestamp objects to be compared in JavaScript using the
+  // arithmetic comparison operators, such as < and >.
+  // https://github.com/firebase/firebase-js-sdk/issues/2632
+  valueOf(): string {
+    const formattedSeconds = Timestamp.normalizeAndPad(
+      this.seconds,
+      Timestamp.MIN_SECONDS,
+      Timestamp.MAX_SECONDS
+    );
+    const formattedNanoseconds = Timestamp.normalizeAndPad(
+      this.nanoseconds,
+      Timestamp.MIN_NANOSECONDS,
+      Timestamp.MAX_NANOSECONDS
+    );
+    return formattedSeconds + '.' + formattedNanoseconds;
+  }
+
+  private static normalizeAndPad(
+    value: number,
+    minValue: number,
+    maxValue: number
+  ): string {
+    const padLength = Math.ceil(Math.log10(maxValue - minValue));
```

Comment:
Most `pad()` methods that I am aware of take as an argument the desired length to pad to. I would suggest you do the same thing here: The nanosecond length is always 9, and the MAX_SECONDS value is a constant that you could define as const or via `Math.ceil(Math.log10(MAX_SECONDS)))`.

Reply:
Thanks for the suggestion.  I'll simply hardcode the pad lengths, since there is really no value in computing those constants over and over again.